src/shogun/classifier/NearestCentroid.cpp

```diff
@@ -88,7 +99,28 @@ namespace shogun{
 		m_is_trained=true;
 		distance->init(centroids_feats, distance->get_rhs());
 
-		return true;
+		return shared_from_this()->as<NearestCentroid>();
 	}
 
+	bool NearestCentroid::train_machine(std::shared_ptr<Features> data)
+	{
+		error("the train_machine have been deprecated, please use fit instead");
```

Comment:
uhoh :) the deprecation should mean that it's still functional :)
so either use a warning and still be functional (but i guess that's not possible coz of not having labels) so in this case i would just drop the whole thing :)
if Machine::train_machine is a pure virtual function, then maybe we should just create there an implementation with this error....